join.go

```diff
@@ -12,6 +12,21 @@ type FederatedJoinClient interface {
 	SendJoin(ctx context.Context, origin, s spec.ServerName, event PDU) (res SendJoinResponse, err error)
 }
 
+type RoomInfo struct {
+	Version RoomVersion
+	NID     int64
```

Comment:
I somewhat feel like that a NID shouldn't be in GMSL. It's more of a concept in Dendrite.